test/internal/compat.py

```diff
@@ -0,0 +1,55 @@
+import sys
+
+# Added for compatibility with Python 3.
+# Python 2 provided the class basestring, from which both str and unicode
+# derived. In Python 3, the concept of basestring makes no sense: text is only
+# represented by str.
+try:
+  basestring
+except NameError:
+  basestring = str
```

Comment:
We don't use `basestring` directly anywhere, only via `string_types`. So we don't use `basestring` when running on Python 3. I removed this chunk of the code and all enabled tests succeeded.

Reply:
Right, removed.